app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -26,28 +26,32 @@ def execute
     input_vars    = options.dig(:input_vars, :extra_vars) || {}
     action        = options[:action]
 
-    case action
-    when ResourceAction::RETIREMENT
-      Terraform::Runner.delete_stack(
-        options[:terraform_stack_id],
-        template_path,
-        :input_vars                  => decrypt_vars(input_vars),
-        :input_vars_type_constraints => input_vars_type_constraints,
-        :credentials                 => credentials,
-        :env_vars                    => options[:env_vars]
-      )
-    else
-      response = Terraform::Runner.create_stack(
-        template_path,
-        :input_vars                  => decrypt_vars(input_vars),
-        :input_vars_type_constraints => input_vars_type_constraints,
-        :credentials                 => credentials,
-        :env_vars                    => options[:env_vars]
+    runner_options = {
+      :input_vars                  => decrypt_vars(input_vars),
+      :input_vars_type_constraints => input_vars_type_constraints,
+      :credentials                 => credentials,
+      :env_vars                    => options[:env_vars]
+    }
+
+    # required for ResourceAction::RECONFIGURE, ResourceAction::RETIREMENT
+    unless action == ResourceAction::PROVISION
+      runner_options = runner_options.merge(
+        :stack_id => options[:terraform_stack_id]
       )
```

Comment:
So two things, first if you just want to set a single key this is better:
```suggestion
      runner_options[:stack_id] = options[:terraform_stack_id]
      )
```

Second, if you did in the future want to merge a hash into an existing hash you can use the `!` version of merge:
`runner_options.merge!(:a => "a", :b => "b")` which is equivalent to `runner_options = runner_options.merge(:a => "a", :b => "b")`

Reply:
Won't hold this PR up for this since this is functional but can you fix this in a follow-up?